src/main/fc/fc_rc.c

```diff
@@ -548,7 +552,27 @@ FAST_CODE void processRcCommand(void)
 #if defined(SIMULATOR_BUILD)
 #pragma GCC diagnostic pop
 #endif
-            calculateSetpointRate(axis);
+            bool useGpsRescueYaw = false;
+#ifdef USE_GPS_RESCUE
+            // Logic improvements to not override rcCommand when GPS Rescue is active.
+            // Instead only modify the flight control values derived from rcCommand
+            // like setpointRate[] and rcDeflection[].
+            if ((axis == FD_YAW) && FLIGHT_MODE(GPS_RESCUE_MODE)) {
+                // If GPS Rescue is active then override the yaw commanded input
+                // with a zero (centered) stick input. Otherwise the pilot's yaw inputs
+                // will get calculated into the rcDeflection values affecting downstream logic
+                useGpsRescueYaw = true;
+            }
+#endif
+            calculateSetpointRate(axis, useGpsRescueYaw);
+
+#ifdef USE_GPS_RESCUE
+            if (useGpsRescueYaw) {
+                // If GPS Rescue is active then override the setpointRate used in the
+                // pid controller with the value calculated from the desired heading logic.
+                setpointRate[axis] = gpsRescueGetYawRate();
```

Comment:
How about

```
￼#ifdef USE_GPS_RESCUE
   if (FLIGHT_MODE(GPS_RESCUE_MODE) && (axis == FD_YAW)) {
        setpointRate[axis] = gpsRescueGetYawRate();
    } else 
#endif
    {
        calculateSetpointRate(axis);
    }
￼ ```
￼

Reply:
Yeah, I already improved that and moved the logic into the `calculateSetpointRate()` function doing a identical `#ifdef` logic.  We also need to zero out the `rcDeflection[]` and `rcDeflectionAbs[]` values.  So it looks like this now:
```
static void calculateSetpointRate(int axis)
{
    float angleRate;
    
#ifdef USE_GPS_RESCUE
    if ((axis == FD_YAW) && IS_RC_MODE_ACTIVE(BOXVTXPITMODE)) {
        // If GPS Rescue is active then override the setpointRate used in the
        // pid controller with the value calculated from the desired heading logic.
        angleRate = gpsRescueGetYawRate();

        // Treat the stick input as centered to avoid any stick deflection base modifications (like acceleration limit)
        rcDeflection[axis] = 0;
        rcDeflectionAbs[axis] = 0;
    } else
#endif
    {
        // scale rcCommandf to range [-1.0, 1.0]
        float rcCommandf = rcCommand[axis] / 500.0f;
        rcDeflection[axis] = rcCommandf;
        const float rcCommandfAbs = ABS(rcCommandf);
        rcDeflectionAbs[axis] = rcCommandfAbs;

        angleRate = applyRates(axis, rcCommandf, rcCommandfAbs);
    }
    setpointRate[axis] = constrainf(angleRate, -SETPOINT_RATE_LIMIT, SETPOINT_RATE_LIMIT); // Rate limit protection (deg/sec)

    DEBUG_SET(DEBUG_ANGLERATE, axis, angleRate);
}
```

And the logic in `processRcCommand()` is back to the way it started with simply calling `calculateSetpointRate(axis)`.

Reply:
I hope not like _that_:

Reply:
Sorry, that's my debug testing.  :)  No, that won't be in the commit.

Reply:
@etracer65 : Much better than before, thanks ... 

Bigger redesign of control/data flow would be nice eventually ...